### PR TITLE
server/auth: enable tokenProvider if recoved store enables auth

### PR DESCRIPTION
**Pull request by cfz**
we found a lease leak issue:
if a new member(by member add) is recovered by snapshot, and then
become leader, the lease will never expire afterwards. leader will
log the revoke failure caused by "invalid auth token", since the
token provider is not functional, and drops all generated token
from upper layer, which in this case, is the lease revoking
routine.

### CODE REVIEW

**Comment by mitake** [comment]
@cfz thanks a lot for this PR! I think it will fix the issue, but let me confirm with it. I think I can submit my review on Sunday.

**Review by mitake — APPROVED** [review]
LGTM, thanks! BTW it's also great if you can add an e2e test case. Can you add it? If you aren't familiar with e2e test infra, I can take it.

**Comment by mitake** [comment]
On the second thought, the change improves simple token and it's not for production use cases. I think having e2e test case is too much. It would be good just merging it.

**Comment by cfz** [comment]
> LGTM, thanks! BTW it's also great if you can add an e2e test case. Can you add it? If you aren't familiar with e2e test infra, I can take it.

i can have a try :)

**Comment by cfz** [comment]
> On the second thought, the change improves simple token and it's not for production use cases. I think having e2e test case is too much. It would be good just merging it.

that's also what i'm thinking about.

actually, we enabled auth but using Common Name based authentication for client side. looks like we eventually run into the simple token code somehow....

**Comment by cfz** [comment]
i would also like to backport this change to v3.4, which is the actual release we are using. but not sure we can do this without a new commit, due to the path changing during 3.4 and 3.5

**Comment by cfz** [comment]
seems the workflows needs another approval, since i amend my commit.... @mitake

**Comment by cfz** [comment]
ping @mitake 😄

**Comment by mitake** [comment]
@cfz approved, thanks!

File: server/auth/simple_token.go
Comment (spzala) [inline server/auth/simple_token.go:164]: Thanks @cfz one comment here - I think you should put the defer statement before the if loop and remove the Unlock statement from the if loop. Thanks! cc @mitake

Comment (cfz) [reply server/auth/simple_token.go]: good comment @spzala. updated

Comment (spzala) [reply server/auth/simple_token.go]: @cfz thanks for addressing it quickly! A nit now - can you please remove the blank lines on 161 and 165 for better readability. Thanks!

Comment (cfz) [reply server/auth/simple_token.go]: done @spzala

**Comment by spzala** [comment]
@cfz thanks for addressing my comments, lgtm.

**Review by spzala — APPROVED** [review]

**Comment by mitake** [comment]
merged, thanks @cfz @spzala !

**Pull request merged**
